File: articles/active-imagination-and-the-practice-of-symbolic-dialogue/fortran/dialogue_integration.f90
! Active Imagination and the Practice of Symbolic Dialogue
! Fortran example: simple synthetic symbolic-integration score

program dialogue_integration_demo
  implicit none

  real :: risk
  real :: score
  real :: imbalance
  real :: positive_risk

  risk = 0.54 * 0.56 - 0.38 * 0.88 - 0.34 * 0.88
  imbalance = 0.88 - 0.56
  positive_risk = max(risk, 0.0)

  score = 0.56 * 0.88 + 0.52 * 0.56 + 0.48 * 0.88 + &
          0.44 * 0.90 + 0.34 * 0.86 - 0.70 * imbalance * imbalance - &
          0.28 * positive_risk

  print *, "Synthetic destabilization risk:", risk
  print *, "Synthetic symbolic integration score:", score
end program dialogue_integration_demo

File: articles/active-imagination-and-the-practice-of-symbolic-dialogue/go/dialogue_integration.go
package main

import "fmt"

func destabilizationRisk(
	imaginalActivation,
	egoMediation,
	reflectiveResponse float64,
) float64 {
	return 0.54*imaginalActivation -
		0.38*egoMediation -
		0.34*reflectiveResponse
}

func symbolicIntegration(
	egoMediation,
	imaginalActivation,
	reflectiveResponse,
	symbolicRelation,
	ethicalContainment,
	destabilization float64,
) float64 {
	imbalance := egoMediation - imaginalActivation
	positiveDestabilization := destabilization
	if positiveDestabilization < 0 {
		positiveDestabilization = 0
	}

	return 0.56*egoMediation +
		0.52*imaginalActivation +
		0.48*reflectiveResponse +
		0.44*symbolicRelation +
		0.34*ethicalContainment -
		0.70*imbalance*imbalance -
		0.28*positiveDestabilization
}

func main() {
	risk := destabilizationRisk(0.56, 0.88, 0.88)
	score := symbolicIntegration(0.88, 0.56, 0.88, 0.90, 0.86, risk)

	fmt.Printf("Synthetic destabilization risk: %.3f\n", risk)
	fmt.Printf("Synthetic symbolic integration score: %.3f\n", score)
}
